tests/Unit/Inc/Yoast_Dynamic_Rewrites_Test.php

```diff
@@ -0,0 +1,290 @@
+
+use Mockery;
+use Brain\Monkey;
+use Yoast_Dynamic_Rewrites;
+use RuntimeException;
+use Yoast\WP\SEO\Tests\Unit\TestCase;
+
+
+/**
+ * Unit Test Class.
+ *
+ * @coversDefaultClass Yoast_Dynamic_Rewrites
+ */
+class Yoast_Dynamic_Rewrites_Test extends TestCase {
+
+	/**
+	 * Holds the rewrite class.
+	 *
+	 * @var WP_Rewrite
+	 */
+	private $wp_rewrite;
+
+	/**
+	 * Holds the instance of the class being tested.
+	 *
+	 * @var Yoast_Dynamic_Rewrites
+	 */
+	private $instance;
+
+	/**
+	 * Set up the class which will be tested.
+	 */
+	protected function set_up() {
+		parent::set_up();
+
+		$this->stubTranslationFunctions();
+
+		$this->wp_rewrite        = Mockery::mock( 'WP_Rewrite' );
+		$this->wp_rewrite->index = 'index.php';
+
+		$this->instance = new Yoast_Dynamic_Rewrites( $this->wp_rewrite );
+	}
+
+	/**
+	 * Tear down the class which was tested.
+	 */
+	protected function tear_down() {
+		parent::tear_down();
+
+		unset( $GLOBALS['wp_rewrite'] );
+	}
+
+	/**
+	 * Test construct method.
+	 *
+	 * @covers ::__construct
+	 */
+	public function test_construct_with_runtime_exception() {
+		$this->expectException( RuntimeException::class );
+		$GLOBALS['wp_rewrite'] = null;
+		$instance = new Yoast_Dynamic_Rewrites();
+
+		$this->assertNotEmpty( $instance->wp_rewrite, 'WP_Rewrite instance is set.' );
+		$this->assertInstanceOf( 'WP_Rewrite', $instance->wp_rewrite, 'WP_Rewrite instance is set with the given parameter.' );
+	}
+
+	/**
+	 * Test construct method.
+	 *
+	 * @covers ::__construct
+	 */
+	public function test_construct() {
+		$GLOBALS['wp_rewrite'] = $this->wp_rewrite;
+		$instance              = new Yoast_Dynamic_Rewrites();
+
+		$this->assertNotEmpty( $instance->wp_rewrite, 'WP_Rewrite instance is set.' );
+		$this->assertInstanceOf( 'WP_Rewrite', $instance->wp_rewrite, 'WP_Rewrite instance is set with the given parameter.' );
+	}
+
+	/**
+	 * Tests instance method.
+	 *
+	 * @covers ::instance
+	 */
+	public function test_instance() {
+		global $wp_rewrite;
+		$wp_rewrite            = Mockery::mock( 'WP_Rewrite' );
+		$GLOBALS['wp_rewrite'] = $wp_rewrite;
+		$this->assertInstanceOf( 'Yoast_Dynamic_Rewrites', Yoast_Dynamic_Rewrites::instance(), 'Instance is returned.' );
+		$this->assertSame( Yoast_Dynamic_Rewrites::instance(), Yoast_Dynamic_Rewrites::instance(), 'Same instance is returned.' );
+	}
+
+	/**
+	 * Test register_hooks method.
+	 *
+	 * @covers ::register_hooks
+	 */
+	public function test_register_hooks() {
+		$this->instance->register_hooks();
+
+		$this->assertNotFalse( \has_action( 'init', [ $this->instance, 'trigger_dynamic_rewrite_rules_hook' ] ), 'Register action for dynamic rewrites.' );
+		$this->assertNotFalse( \has_filter( 'option_rewrite_rules', [ $this->instance, 'filter_rewrite_rules_option' ] ), 'Adds option_rewrite_rules.' );
+		$this->assertNotFalse( \has_filter( 'sanitize_option_rewrite_rules', [ $this->instance, 'sanitize_rewrite_rules_option' ] ), 'Adds sanitize_option_rewrite_rules.' );
```

Comment:
The messages are intended to be displayed in case of error, so I'd turn them into something like "Does not have expected init action" etc.